backend/services/monthly_dump_service.py

```
@@ -0,0 +1,350 @@
+            if response.history:
+                for r in response.history:
+                    if not MonthlyDumpService._is_safe_url(r.url):
+                        logger.warning("Rejected disallowed redirect URL", {"url": r.url})
+                        return None
+            
+            response.raise_for_status()
+
+            # Header check for size
+            cl = response.headers.get("Content-Length")
+            if cl and int(cl) > MonthlyDumpService.MAX_IMAGE_BYTES:
+                logger.error("Image exceeds size limit (from header)", {"url": url, "size": cl})
```

Ruff (0.15.10)

[error] Too many arguments for `logging` format string

(PLE1205)